### DISCUSSION TITLE
Ios flutter app code signing issue

### DISCUSSION THREAD

**Discussion by omar-chaabouni** [codemagic-ci-cd/codemagic-docs#1932] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I was trying to build ios flutter app. I have code signing enabled ( with app store integration) but I always got an error during the signing step.

### Expected results

successful code signing

### Actual results

Found 3 Signing Certificates matching specified filters: certificateType=IOS_DISTRIBUTION,DISTRIBUTION.
Did not find any Signing Certificates for given private key
Creating new Signing Certificate: certificate type: DISTRIBUTION
POST https://api.appstoreconnect.apple.com/v1/certificates returned 409: There is a problem with the request entity - You already have a current Distribution certificate or a pending certificate request.
I have automatic ios code signing enabled

**Comment by omar-chaabouni** [comment, marked as answer]
The solution : login to your apple developer account and navigate to this link https://developer.apple.com/account/resources/certificates/list
here you should find your available certificates.
For my app Codemagic actually creates a certificate with type distribution for each codemagic applicaiton I have.
I already had 3 applications (2 were unused) and I just created a new app and I was trying to sign it.
Codemagic already created 3 certificates with type "Distribution" for my other apps, so when I tried to create one now it gave this error.
The reason is that you are only allowed 3 certificates from the same type. (Distribution in this case)
so I deleted the certificates of my unsued apps and when I rerun the workflow, codemagic now creates a new certificate for my newly created app (created only once per application in the first build and used forever) and my code signing works.

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/55398565/8eca6a95-cb28-4905-b761-5dd3a8f0dad1)

**Reply by marianotella** [reply]
Thanks, works great for me
